src/Wrapper/LambdaWrapper.js

```diff
@@ -31,23 +49,15 @@ export default ((configuration, handler) => {
     }
 
     try {
-      return handler.call(instance, di, request, callback);
-    } catch (error) {
-      logger.error(error);
+      let outcome = handler.call(instance, di, request, callback);
 
-      const responseDetails = {
-        body: error.body || {},
-        code: error.code || 500,
-        message: 'unknown error',
-      };
+      if (outcome instanceof Promise) {
+        outcome = outcome.catch(handleError);
+      }
```

Comment:
Looks weird to me that we call the handler, which might end up in a rejected promise, before chaining the `.catch` ... but guessing it's no different from doing `handler().catch()` if we already knew the handler was a promise?

Reply:
Yeah there is no way to know whether the handler is an async function before calling it so if we are handling a normal function chaining `.catch` would throw

Reply:
Cool. Apparently `handler.constructor.name === 'AsyncFunction` if it's async

Reply:
I didn't know that! Cheers

I guess that armed with this knowledge we could wrap it in a if/else and do the try/catch block in one branch and the promise catch in the other. But to be honest I think it would be even uglier! 😄